### ISSUE TITLE
Create standard ABC for wrapped Java objects

### ISSUE THREAD

**Issue by devinrsmith** [deephaven/deephaven-core#1892]
We need a standard interface that all python wrapped Java objects can implement to make it easier for downstream code to unwrap.

Currently, `pyintegration/deephaven2/server/script_session/__init__.py` / `unwrap_to_java_type` code looks like:

```python
    if isinstance(object, deephaven2.table.Table):
        return object.j_table
    if isinstance(object, deephaven.Plot.figure_wrapper.FigureWrapper):
        return object.figure
````

As we wrap more types, a generic ABC for wrapped Java types would make this sort of logic much easier. For example:

```python
    if isinstance(object, JavaWrappedABC):
        return object.j_object
```

**Comment by devinrsmith** [comment]
Chatting w/ Jianfeng, maybe something like:

```python
class DHWrappedABC(ABC):
    @property
    @abc.abstractmethod
    def j_object(self) -> JType:
        pass
```

**Comment by jmao-denver** [comment]
closed by #1825
